src/Microsoft.OpenApi/Models/References/OpenApiTagReference.cs

```diff
@@ -58,6 +59,35 @@ public string? Description
 
         /// <inheritdoc/>
         public string? Name { get => Target?.Name ?? Reference?.Id; }
+
+        /// <inheritdoc/>
+        public string? Summary => Target?.Summary;
+
+        /// <inheritdoc/>
+        public OpenApiTagReference? Parent 
+        { 
+            get 
+            {
+                // Prevent stack overflow by checking for self-reference
+                var targetParent = Target?.Parent;
+                if (targetParent == null)
+                    return null;
+                
+                // Check if the target's parent reference is the same as this reference
+                if (ReferenceEquals(targetParent.Reference, this.Reference))
+                    return null;
+                
+                // Check if the target's parent name is the same as this tag's name
+                if (string.Equals(targetParent.Name, this.Name, StringComparison.OrdinalIgnoreCase))
+                    return null;
```

Comment:
if the Target is a reference, and reference equals this, return null, otherwise, return the parent